### DISCUSSION TITLE
Unable to use github ssh for anything (pull/clone/push etc)

### DISCUSSION THREAD

**Discussion by acharyarupak391** [community/community#54558] [Repositories] [answered]
### Select Topic Area

General

### Body

I have setup ssh in my local machine & in my github account setting.
It was working fine until yesterday or the day before, but suddenly today, it stopped working.

But when I try to do anything(fetch/pull/push/clone), it always gives this response:
```
ssh: connect to host github.com port 22: Operation timed out
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I thought if there was any issue for my device to connect with the ssh server 
so i ran `telnet github.com 22` and got this response:
```
Trying 20.205.243.166...
telnet: connect to address 20.205.243.166: Operation timed out
telnet: Unable to connect to remote host
```

I'm not sure if there's any issue on my end, or from the server itself(which is very unlikely),
so I'm really confused.

I regenerated the ssh key and added the new key to my github account. But the problem still persists.

Any idea what should I do?

**UPDTE:** The problem seems to be in my specific ISP for some reason. Switching to a different Wi-Fi worked. Also, port `443` seems to work, but no connection with `22`.

**Comment by airtower-luna** [comment]
Timeout means it's definitely not a problem with your key. Most likely it means a misconfigured firewall *somewhere* that drops your packets, or maybe a router that sends them in a very wrong direction.

Things to try (other than checking your firewall settings):
* `traceroute -T -p 22 github.com` to see *where* the packets get lost.
* Try [using SSH over the HTTPS port](https://docs.github.com/en/authentication/troubleshooting-ssh/using-ssh-over-the-https-port). This might be a workaround in case the responsible party is intentionally blocking SSH (bad!) or simply slow to fix the problem.

**Comment by acharyarupak391** [comment]
@airtower-luna 
that exact command doesn't work but `traceroute -p 22 github.com` works and gives something like this:
(replaced some parts with `x`s. not sure how sensitive & vulnerable it is)
```
1  192.168.xxx.xxx (192.168.xxx.xxx)  3.420 ms  2.427 ms  7.211 ms
2  202.94.xxx.xxx (202.94.xxx.xxx)  4.669 ms  4.448 ms  4.037 ms
3  202.94.xxx.xxx (202.94.xxx.xxx)  7.431 ms  9.849 ms  8.054 ms
4  202.94.xxx.xxx (202.94.xxx.xxx)  29.759 ms  8.679 ms  7.207 ms
5  49.236.xxx.xxx (49.236.xxx.xxx)  8.714 ms  7.522 ms  9.822 ms
6  125.17.xxx.xxx (125.17.xxx.xxx)  9.660 ms  8.108 ms  12.667 ms
7  116.119.xxx.xxx (116.119.xxx.xxx)  49.040 ms  47.684 ms
    116.119.xxx.xxx (116.119.xxx.xxx)  48.124 ms
8  ae60-0.xxx.xxx.xxx (198.200.130.17)  47.068 ms  49.237 ms  53.990 ms
9  ae21-0.xxx.xxx.xxx (104.44.41.235)  48.384 ms  48.545 ms  48.796 ms
10 be-23-0.xxx.xxx.xxx (104.44.22.127)  83.707 ms  84.523 ms x
11 be-9-0.xxx.xxx.xxx (104.44.18.201)  78.160 ms  177.695 ms
    be-11-0.xxx.xxx.xxx (104.44.18.199)  84.591 ms
12 ae122-0.xxx.xxx.xxx (104.44.20.18)  79.825 ms
    ae102-0.xxx.xxx.xxx (104.44.20.2)  83.894 ms  79.521 ms
13 * * *
14 * * *
15 * * *
16 * * *
17 * * *
18 * * *
19 * * *
20 * * *
```

Also, another important discovery on this whole thing, 
## it works fine when I switch my Wi-Fi network
I have no idea why my ISP would restrict connecting to github's ssh server, though `http` works fine.
I'm even more confused 🫤

**Comment by airtower-luna** [comment]
> I have no idea why my ISP would restrict connecting to github's ssh server, though http works fine.

Yeah, that sounds a lot like a network issue. I can't say if they restrict that intentionally or if it's a bug, but sending a complaint/bug report is probably a good idea.

I'm afraid other than that the options you have are to use another network, or try to work around the problem (e.g. using SSH over the HTTPS port as mentioned above, using Git over HTTPS, or using a VPN).

**Reply by acharyarupak391** [reply]
Thanks for the help.

> using SSH over the HTTPS port as mentioned above

How to do this? 
I've set up ssh in my machine & in my github account. Doesn't this mean I'm already using ssh instead of https for all git actions?

**Reply by airtower-luna** [reply, marked as answer]
You are already using SSH, which usually uses TCP port 22. Networks that do weird blocks usually still allow TCP port 443, which is the default port for HTTPS. Because of that GitHub offers an SSH server that listens on `ssh.github.com` TCP port 443, which gets around simple port blocks (a more sophisticated block might notice the protocol is SSH, not TLS). :wink: 

See [Use SSH over HTTPS port](https://docs.github.com/en/authentication/troubleshooting-ssh/using-ssh-over-the-https-port) in the documentation. Though I'd suggest using [`url.<base>.insteadOf`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf) setting *instead of* overriding the connection for `github.com` in `~/.ssh/config` (that's what the documentation describes), like so:

```
git config --global "url.ssh://git@ssh.github.com:443/.insteadOf" git@github.com:
```

The resulting entry in your `~/.gitconfig` should look like this (just remove it if you want to go back to SSH over port 22):

```
[url "ssh://git@ssh.github.com:443/"]
	insteadOf = git@github.com:
```

The reason is that this setting is specific to how Git uses SSH, not SSH in general.

**Reply by acharyarupak391** [reply]
Great, doing `ssh -T git@ssh.github.com -p 443` works. I guess port `443` works but `22` doesn't. 
idk why there's only issue with port 22 with my ISP 🤔
I've informed their dev about this, hope that gets solved.

**OK, that solved the issue.** 
It feels a bit slow compared to before but that's better than using HTTPS with personal access token 😂
Let's hope it doesn't bring other issues 🥲

Thanks a lot @airtower-luna. You're a saver 🙏 😊

**Reply by airtower-luna** [reply]
Glad it works, and you're welcome! :blush:

**Reply by kyuuube** [reply]
Bro！ You're a saver！

**Reply by remcovdsteen** [reply]
Thanks for this solution! Awesome.

**Reply by Kharzette** [reply]
This started for me this morning.  The last hop I can get to on 22 is in twelve99.net around Reston Virginia.

**Comment by Ai-Ya-Ya** [comment]
Is there a third port we can use? Both 22 and 443 fail for me. My `traceroute` looks like achary's but with only two non-starred paths.

**Reply by Shildeburger** [reply]
same problem for me
